spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -22,18 +22,19 @@
   end
 
   describe "::call" do
-    it "send all forms of welcome notifications" do
-      user = create(:user, :with_identity, identities: ["github"], created_at: 1.week.ago)
-      sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
-      expect(user.notifications.count).to eq(2)
-    end
-
     it "does not send a notification to an unsubscribed user" do
       user = create(:user, :with_identity, identities: ["github"], created_at: 1.week.ago, welcome_notifications: false)
       expect do
         sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
       end.to not_change(user.notifications, :count)
     end
+
+    it "send only 1 notifications at a time" do
+      user = create(:user, :with_identity, identities: ["github"], created_at: 1.week.ago)
+      expect do
+        sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
+      end.to change(user.notifications, :count).by(1)
```

Comment:
Should we check _which_ notification they are/are not getting here? I'm assuming it should be the notification for the `welcome_broadcast` here, and not the `authentication_broadcast` (even though they qualify for the authentication_broadcast`).

Maybe the test could test that the first time you send call the generator, they get the `welcome_broadcast` notification, and the second time, they get the `authentication_broadcast`?

Reply:
Sorry I keep putting this spec off. I will update this portion with the next set of notification implementation.